### PR TITLE
autotest: add test for copter position reset

### PR DESCRIPTION
**Pull request by peterbarker**
Reproduces https://github.com/ArduPilot/ardupilot/issues/14236

image

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -134,7 +134,10 @@ def takeoff(self,
             self.wait_ready_to_arm(require_absolute=require_absolute)
             self.zero_throttle()
             self.arm_vehicle()
-        self.set_rc(3, takeoff_throttle)
+        if mode == 'GUIDED':
```
Comment (khancyr) [inline Tools/autotest/arducopter.py:137]: this can now be used in more place !

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -4795,6 +4798,33 @@ def test_parameter_validation(self):
         self.drain_mav()
         self.assert_prearm_failure("Check MOT_PWM_MIN/MAX")
 
+    def test_copter_gps_zero(self):
+        self.progress("Starting....2")
```
Comment (khancyr) [inline Tools/autotest/arducopter.py:4802]: starting 2 ?

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -4795,6 +4798,33 @@ def test_parameter_validation(self):
...
+
+        # https://github.com/ArduPilot/ardupilot/issues/14236
+        self.progress("arm the vehicle and takeoff in Guided")
+        self.takeoff(20, mode='GUIDED')
+        self.progress("fly 50m North (or whatever)")
+        old_pos = self.mav.recv_match(type='GLOBAL_POSITION_INT', blocking=True)
+        self.fly_guided_move_global_relative_alt(50,0, 20)
```
Comment (khancyr) [inline Tools/autotest/arducopter.py:4809]: 50, 0

File: Tools/autotest/arducopter.py [hunk 4]
```diff
@@ -4795,6 +4798,33 @@ def test_parameter_validation(self):
...
+        self.progress("param set GPS_TYPE 0")
+        self.set_parameter('GPS_TYPE', 0)
```
Comment (khancyr) [inline Tools/autotest/arducopter.py:4811]: maybe wait_gps_disable() ?

**Comment by khancyr** [comment]
is this mean to be merge ?

**Comment by peterbarker** [comment]
@khancyr this shows a bug we currently have in master.  It currently actually ensures we *have* the bug; it will need some adjustment to actually ensure we *don't* have the bug in the future :-)

 AFAIK @rmackay9 is still working through the bug itself.

Thanks for the review.

**Comment by khancyr** [comment]
yep, I notice it was liked to Randy PR but your wording on this one was weird, that why I ask !

**Comment by peterbarker** [comment]
I've rejigged this to fail if we see the same bug reappear in Copter.

I'm not going to merge this just yet; we need to come to grips with the amount of time our tests are taking in CI, and perhaps run a subset of tests in there.  I'm currently thinking we create a `CopterTests3` which is run when you do a `test.Copter` but it NOT run on CI (at least until we have more resources there).  These tests may fail on autotest - @tridge suggested we send a notification to Discord whenever an autotest run completes.

**Comment by khancyr** [comment]
@peterbarker we now have better CI, wanna pursuit this ?

**Comment by peterbarker** [comment]
@khancyr did it up a little bit.

**Pull request merged**